e2e/components/TagSet/TagSet-test.avt.e2e.js

```diff
@@ -21,4 +22,193 @@ test.describe('TagSet @avt', () => {
+      .click();
+
+    await modalElement.evaluate((element) =>
+      Promise.all(
+        element.getAnimations().map((animation) => animation.finished)
+      )
+    );
+
+    await expect(page.getByRole('heading', { name: 'All tags' })).toBeVisible();
+
+    //checking search funtionality
+    await expect(page.getByRole('searchbox')).toBeFocused();
+    await expect(modalElement.getByText('Two')).toBeVisible();
+    await page.getByRole('searchbox').fill('One');
+    await expect(modalElement.getByText('Two')).toBeHidden();
+
+    await page.getByRole('button', { name: 'Close' }).click();
+    await expect(page.locator(`.${carbon.prefix}--modal`)).toBeHidden();
+  });
+
+  test('@avt-close-tags', async ({ page }) => {
+    await visitStory(page, {
+      component: 'TagSet',
+      id: 'ibm-products-components-tag-set-tagset--with-close-and-overflow-tags',
+      globals: {
+        carbonTheme: 'white',
+      },
+    });
+
+    await expect(
+      page.locator(`.${pkg.prefix}--tag-set__displayed-tag`).getByText('One')
+    ).toBeVisible();
+    await page.getByRole('button', { name: 'Dismiss' }).first().click();
+    await expect(
+      page.locator(`.${pkg.prefix}--tag-set__displayed-tag`).getByText('One')
+    ).toBeHidden();
+  });
+
+  test('@avt-keyboard', async ({ page }) => {
+    await visitStory(page, {
+      component: 'TagSet',
+      id: 'ibm-products-components-tag-set-tagset--with-close-and-overflow-tags',
+      globals: {
+        carbonTheme: 'white',
+      },
+    });
+    const modalElement = page.locator(`.${carbon.prefix}--modal.is-visible`);
+
+    await page.keyboard.press('Tab');
+    await expect(
+      page.getByRole('button', { name: 'Dismiss' }).first()
+    ).toBeFocused();
+    pressTabKey(page, 6);
+
+    const moreTagsButton = page.locator(
+      `.${pkg.prefix}--tag-set-overflow__popover-trigger`
+    );
+    await expect(moreTagsButton).toBeFocused();
+    await page.keyboard.press('Enter');
+
+    await expect(
+      page.locator(`.${carbon.prefix}--popover--open`)
+    ).toBeVisible();
+
+    pressTabKey(page, 1); //first tag inside popover is focussed
+    await expect(
+      page
+        .locator(`.${pkg.prefix}--tag-set-overflow__tag-list`)
+        .getByRole('button', { name: 'Dismiss' })
+        .first()
+    ).toBeFocused();
+
+    pressTabKey(page, 10);
+
+    await expect(
+      page.locator(`.${pkg.prefix}--tag-set-overflow__show-all-tags-link`)
+    ).toBeFocused(); //view all tags modal link is focussed
+
+    await page.keyboard.press('Enter');
+
+    await modalElement.evaluate((element) =>
+      Promise.all(
+        element.getAnimations().map((animation) => animation.finished)
+      )
+    );
+
+    await expect(page.getByRole('heading', { name: 'All tags' })).toBeVisible();
+    pressTabKey(page, 1);
+    await page.keyboard.press('Enter');
+    await expect(page.locator(`.${carbon.prefix}--modal`)).toBeHidden();
+  });
 });
+async function pressTabKey(page, number) {
```

Comment:
great idea for a utility! i think you should move this into `global/js/utils/test-utils.js` file to make the code more widely available.

Reply:
Hey @davidmenendez ,
I found a existing util folder for e2e test and moved to that location(`e2e/test-utils/simulateKeyPress.ts`). Not sure this is apt as a global util since `page` is relevant in testing aspect only.
What do you think ?